cmd/component-codegen/cmd/root.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright © 2025 NAME HERE <EMAIL ADDRESS>
+*/
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "component-codegen",
+	Short: "A cli app to generate boilerplate code",
+	Long: `A cli which can be used by component teams to generate boiler plate code 
+	while onboarding new components to the opendatahub operator.`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() { //nolint:gochecknoinits
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.tools.yaml)")
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```

Comment:
```suggestion
	//rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```

I might be wrong about this one, but guessing that it'd an example, like L34?

Reply:
Yes, comment was removed by mistake. Added it back as a comment now.